runtime/queries/markdown/highlights.scm

```diff
@@ -1,11 +1,11 @@
 ;From MDeiml/tree-sitter-markdown & Helix
 (setext_heading
   (paragraph) @markup.heading.1
-  (setext_h1_underline) @markup.heading.1)
+  (setext_h1_underline) @markup.heading.1.delimiter)
 
 (setext_heading
   (paragraph) @markup.heading.2
-  (setext_h2_underline) @markup.heading.2)
+  (setext_h2_underline) @markup.heading.2.delimiter)
```

Comment:
Also for markdown headings. Can revert this commit if wanted, but maybe let it bake on master to get hate mail first.

![image](https://github.com/user-attachments/assets/7a3a272d-6534-4dfb-826a-d6d5b6bc97d3)

Reply:
Looks nice, but I would indeed prefer to keep Markdown out of it (for now); the potential impact is too wide. Keeping this focused on vimdoc would be much more manageable (and a clearer improvement).